06_predict_survival/nbconverted/plot_survival_results.py

```diff
@@ -196,12 +209,335 @@
+all_data_results_df.head()
+
+
+# In[22]:
+
+
+all_data_compare_df = au.compare_control_ind(all_data_results_df,
+                                       identifier='identifier',
+                                       metric='cindex',
+                                       verbose=True)
+all_data_compare_df['cancer_type'] = all_data_compare_df.identifier.str.split('_', 1, expand=True)[0]
+all_data_compare_df['training_data'] = all_data_compare_df.identifier.str.split('_', 1, expand=True)[1]
+
+print(len(all_data_compare_df))
+all_data_compare_df.head()
+
+
+# In[23]:
+
+
+sns.set({'figure.figsize': (18, 8)})
+    
+all_data_compare_df.sort_values(by='cancer_type', inplace=True)
+
+sns.boxplot(data=all_data_compare_df[~all_data_compare_df.cancer_type.isin(drop_cancer_types)],
+            x='cancer_type',
+            y='delta_cindex',
+            hue='training_data',
+            hue_order=sorted(all_data_compare_df.training_data.unique()))
+plt.xlabel('TCGA cancer type')
+plt.ylabel('cindex(signal) - cindex(shuffled)')
+plt.title('Survival prediction, {}'.format(all_results_desc))
+plt.ylim(-0.5, 0.5)
+    
+plt.tight_layout()
+
+
+# ### Heatmap, all data types
+
+# In[24]:
+
+
+all_data_all_results_df = au.compare_all_data_types(all_data_results_df[~all_data_results_df.cancer_type.isin(drop_cancer_types)],
+                                                    SIG_ALPHA,
+                                                    identifier='cancer_type',
+                                                    metric='cindex')
+
+all_data_all_results_df.rename(columns={'gene': 'cancer_type'}, inplace=True)
+all_data_all_results_df.sort_values(by='p_value').head(10)
+
+
+# In[25]:
+
+
+all_data_heatmap_df = (all_data_all_results_df
+    .pivot(index='training_data', columns='cancer_type', values='delta_mean')
+    .reindex(sorted(all_data_compare_df.training_data.unique()))
+)
+all_data_heatmap_df.iloc[:, :5]
+
+
+# In[26]:
+
+
+raw_results_df = (all_data_results_df
+    .drop(columns=['identifier'])
+    .rename(columns={'cancer_type': 'identifier'})
+)
+raw_results_df.head()
+
+
+# In[27]:
+
+
+sns.set({'figure.figsize': (28, 6)})
+sns.set_context('notebook', font_scale=1.5)
+
+ax = plu.plot_heatmap(all_data_heatmap_df,
+                      all_data_all_results_df.reset_index(drop=True),
                       different_from_best=True,
                       raw_results_df=raw_results_df,
                       metric='cindex',
-                      id_name='cancer_type')
+                      id_name='cancer_type',
+                      scale=(-0.1, 0.4))
+
+plt.title('Performance by cancer type for survival prediction, {}'.format(all_results_desc), pad=15)
+if SAVE_FIGS:
+    images_dir.mkdir(exist_ok=True)
+    plt.savefig(images_dir / 'survival_all_heatmap.svg', bbox_inches='tight')
+    plt.savefig(images_dir / 'survival_all_heatmap.png',
+                dpi=300, bbox_inches='tight')
+
```

Comment:
I probably missed it somewhere, but what is the difference between blue and red boxes?

Reply:
Blue = significantly better than label-permuted baseline, but significantly worse than best-performing data type
Red = significantly better than label-permuted baseline, and not significantly different from best-performing data type (i.e. "statistically equivalent to best")
No box = not significantly better than label-permuted baseline

This is in the paper already, but I'll add it to the notebook too - I don't think it's described anywhere in the code.